frontend/src/metabase/parameters/components/ValuesSourceSettings/ValuesSourceSettings.tsx

```diff
@@ -62,38 +64,68 @@ const ValuesSourceSettings = ({
   );
 };
 
+function isOnlyConnectedFieldSourceAllowed(parameter: Parameter) {
```

Comment:
The name and the function body is a bit hard to follow. I'd change it to something like:
```
function hasLinkedFilters({ filteringParameters }: Parameter) {
  return filteringParameters != null && filteringParameters.length > 0;
}
```

and later use it like this:
```
isEditDisabled: hasLinkedFilters(parameter)
```